app/services/taxonomy/publish_taxon.rb

```diff
@@ -25,6 +25,14 @@ def publish
       raise(InvalidTaxonError, I18n.t('errors.invalid_taxon'))
     end
 
+    def unpublish(type:, alternative_path: nil)
```

Comment:
Hmm. I'll think about this tomorrow. Inlining seems most reasonable for now, because if I was to create something more generic I'd expect to use it differently in the future anyway (redirecting or withdrawing instead of just `gone`-ing it).

@carvil brain dumping is definitely welcome. I'm sure I'm missing a lot of context here, because I have almost the complete opposite view about the number of services (I count 14, and I probably couldn't tell you what most of them are for)